internal/resources/policy/kind/network/recipe/custom_common_schema.go

```diff
@@ -0,0 +1,452 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package recipe
+
+import (
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipenetworkcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/network/common"
+)
+
+var (
+	port = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			portKey: {
+				Type:        schema.TypeString,
+				Description: "The port on the given protocol. This can either be a numerical or named port on a pod.",
+				Optional:    true,
+			},
+			protocolKey: {
+				Type:        schema.TypeString,
+				Description: "The protocol (TCP or UDP) which traffic must match.",
+				Optional:    true,
+			},
+		},
+	}
+
+	toPodLabel = &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Pod Labels on which traffic should be allowed. Use a label selector to identify the pods to which the policy applies.",
+		Optional:    true,
+		MinItems:    1,
+		Elem:        label,
+	}
+
+	ip = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			ipBlockKey: {
+				Type:        schema.TypeList,
+				Description: "IPBlock defines policy on a particular IPBlock. If this field is set then neither of the namespaceSelector and PodSelector can be set.",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						cidrKey: {
+							Type:        schema.TypeString,
+							Description: "CIDR is a string representing the IP Block Valid examples are \"192.168.1.1/24\" or \"2001:db9::/64\"",
+							Required:    true,
```

Comment:
No, it doesn't complain since we are using the `oneOf` way. Either it would be IP block in the RuleSpec or it would be the Selector block, hence those are optional. But in case of IP block, `CIDR ` is a required field.